### PR TITLE
Readmes rach java8

### PR DESCRIPTION
**Pull request by Rachael-E**
Updated readmes here for: 

TimeBasedQuery
TokenAuthentication
FeatureLayerUpdateAttributes
FeatureLayerUpdateGeometries
UpdateRelatedFeatures

@sorenoid please could you do a first review?

### CODE REVIEW

File: java/time-based-query/README.md [hunk 1]
```diff
@@ -1,21 +1,36 @@
...
+Run the sample, and a subset of records will be displayed on the map.
 
 ## How it works
-1. Create an instance of `ServiceFeatureTable` in manual cache mode.
-1. Create a feature layer from the table and add it to the map's operational layers.
-1. Then create an instance of `QueryParameters` and specify a time extent with a start and end time. 
-1. Finally, use `populateFromServiceAsync(query, clearCache, outFields)` method present on `ServiceFeatureTable` to populate features based on the specified time interval.
+
+1. An instance of `ServiceFeatureTable` is created by passing a URL to the REST endpoint of a time-enabled service. Time-enabled services will have TimeInfo defined in the service description. This information is specified in ArcMap or ArcGIS Pro prior to publishing the service.
+2. Create an instance of `ServiceFeatureTable` in manual cache mode, so that the developer can control how and when the feature table is populated with data.
```
Comment (sorenoid) [inline java/time-based-query/README.md]: mixed up grammar here.
```suggestion
2. Create an instance of `ServiceFeatureTable` in manual cache mode, so you can control how and when the feature table is populated with data.
```

File: java/update-related-features/README.md [hunk 2]
```diff
@@ -1,22 +1,43 @@
...
 ## How it works
-When you tap on the map the app identifies if a feature is selected and queries for related features on its `FeatureTable`.  Results are shown in an editable `Callout` where you can update the visitor amount by selecting a new value from the drop-down list.  
 
-Updates can be applied to the server using `ServiceFeatureTable.updateFeatureAsync(...)` and `ServiceFeatureTable.applyEditsAsync()`
+1. Create two `ServiceFeatureTable`s from the Feature Service URLs.
+2. Create two `FeatureLayer`s using the previously created service feature tables.
+3. Add these feature layers to the map.
+4. When a `Feature` is selected, identify and highlight the selected feature.
+5. Retrieve related features by calling `ServiceFeatureTable.queryRelatedFeaturesAsync` and passing in the selected feature.
+6. Updates can be applied to the server using `ServiceFeatureTable.updateFeatureAsync(...)` and `ServiceFeatureTable.applyEditsAsync()`.
```
Comment (sorenoid) [inline java/update-related-features/README.md]: I believe, and am not sure, that `updateFeatureAsync()` updates the local table, and that only `applyEditsAsync()` updates the server.

**Comment by Rachael-E** [comment]
thanks @sorenoid! made those small changes. @hud10837 could you take a second look please, thanks.

File: java/time-based-query/README.md [hunk 3]
```diff
@@ -1,21 +1,36 @@
...
+Run the sample, and a subset of records will be displayed on the map.
 
 ## How it works
-1. Create an instance of `ServiceFeatureTable` in manual cache mode.
-1. Create a feature layer from the table and add it to the map's operational layers.
-1. Then create an instance of `QueryParameters` and specify a time extent with a start and end time. 
-1. Finally, use `populateFromServiceAsync(query, clearCache, outFields)` method present on `ServiceFeatureTable` to populate features based on the specified time interval.
+
+1. An instance of `ServiceFeatureTable` is created by passing a URL to the REST endpoint of a time-enabled service. Time-enabled services will have TimeInfo defined in the service description. This information is specified in ArcMap or ArcGIS Pro prior to publishing the service.
+2. Create an instance of `ServiceFeatureTable` in manual cache mode, so you can control how and when the feature table is populated with data.
```
Comment (hud10837) [inline java/time-based-query/README.md]: I think this step would be better worded as "set the feature request mode of ..." as we have already created the service feature table.

File: java/time-based-query/README.md [hunk 4]
```diff
@@ -1,21 +1,36 @@
...
+
+Run the sample, and a subset of records will be displayed on the map.
 
 ## How it works
-1. Create an instance of `ServiceFeatureTable` in manual cache mode.
-1. Create a feature layer from the table and add it to the map's operational layers.
-1. Then create an instance of `QueryParameters` and specify a time extent with a start and end time. 
-1. Finally, use `populateFromServiceAsync(query, clearCache, outFields)` method present on `ServiceFeatureTable` to populate features based on the specified time interval.
+
+1. An instance of `ServiceFeatureTable` is created by passing a URL to the REST endpoint of a time-enabled service. Time-enabled services will have TimeInfo defined in the service description. This information is specified in ArcMap or ArcGIS Pro prior to publishing the service.
```
Comment (hud10837) [inline java/time-based-query/README.md]: Maybe it's better to word this as an imperative?
```suggestion
1. Create an instance of `ServiceFeatureTable` by passing a URL to the REST endpoint of a time-enabled service. Time-enabled services will have TimeInfo defined in the service description. This information is specified in ArcMap or ArcGIS Pro prior to publishing the service.
```

File: java/token-authentication/README.md [hunk 5]
```diff
@@ -1,36 +1,40 @@
...
 
 ## How it works
 
 1. Create an `AuthenticationChallengeHandler` using the `DefaultAuthenticationChallengeHandler` to handle the challenges sent by the protected map service.
-1. Set the `AuthenticationChallengeHandler` used by the `AuthenticationManager`.
-1. Create a `Portal` to ArcGIS Online.
-1. Create a `PortalItem` for the protected web map using the `Portal` and Item ID of the protected map service.
-1. Create a map to display in the `MapView` using the `PortalItem`.
-1. Set the map to display in the `MapView`.
+2. Set the `AuthenticationChallengeHandler` used by the `AuthenticationManager`.
```
Comment (hud10837) [inline java/token-authentication/README.md]:
```suggestion
2. Set the `AuthenticationChallengeHandler` to the `AuthenticationManager`.
```

File: java/update-related-features/README.md [hunk 6]
```diff
@@ -1,22 +1,43 @@
...
+Update related features in an online feature service.
 
-## How to use the sample
-Tap on a feature to select it. A callout will appear with a spinner which can be used to select new data. Selecting new data will also update the server.
+![Image of update related features](update-related-features.png)
+![Image of map legend](legend.png)
+
+## Use case
+
+Updating related features is a helpful workflow when you have two features with shared or dependent attributes. In a data collection scenario where origin tree features are related to destination inspection records, trees might undergo inspection on some regular interval to assess their size, health, and other characteristics. When logging a new inspection record that captures the latest trunk diameter and condition of a tree, updating these attributes on the origin tree feature would permit the tree point to be symbolized most accurately according to these latest observations.
```
Comment (hud10837) [inline java/update-related-features/README.md]: This seems to be missing a How to use the sample section.

File: java/update-related-features/README.md [hunk 7]
```diff
@@ -1,22 +1,43 @@
...
+# Update related features
 
-![Update Related Features App](update-related-features.png)
-![Map Legend](legend.png)
+Update related features in an online feature service.
 
-## How to use the sample
-Tap on a feature to select it. A callout will appear with a spinner which can be used to select new data. Selecting new data will also update the server.
+![Image of update related features](update-related-features.png)
+![Image of map legend](legend.png)
```
Comment (hud10837) [inline java/update-related-features/README.md]: Should the legend perhaps go in the About the data section?

File: java/update-related-features/README.md [hunk 8]
```diff
@@ -1,22 +1,43 @@
...
+5. Retrieve related features by calling `serviceFeatureTable.queryRelatedFeaturesAsync` and passing in the selected feature.
+6. Update feature in the feature table with `serviceFeatureTable.updateFeatureAsync(selectedFeature)` and apply updates to the server using  `serviceFeatureTable.applyEditsAsync()`.
 
 ## Relevant API
+
 * ArcGISFeature
-* ServiceFeatureTable
-* RelatedQueryParameters
 * RelatedFeatureQueryResult
+* RelatedQueryParameters
```
Comment (hud10837) [inline java/update-related-features/README.md]: I can't find this class in the sample

File: java/update-related-features/README.md (hunk 7 above)
Comment (Rachael-E) [reply java/update-related-features/README.md]: let's just leave it here, it goes with the preview image of the sample well I think.

**Comment by Rachael-E** [comment]
gosh thanks for catching all of those @hud10837! Could you give my changes are a double check please before I merge?

**Pull request merged**